diff_ops_test with CUDA

The following error is given:

```
_______ test_part_deriv[2d- method=central - constant=0 - impl='cuda' ] _______

space = uniform_discr([0.0, 0.0], [1.0, 1.0], [5, 5], impl='cuda')
method = 'central', padding = ('constant', 0)

    def test_part_deriv(space, method, padding):
        """Discretized partial derivative."""

        with pytest.raises(TypeError):
            PartialDerivative(odl.rn(1))

        if isinstance(padding, tuple):
            pad_mode, pad_const = padding
        else:
            pad_mode, pad_const = padding, 0

        # discretized space
        dom_vec = noise_element(space)
        dom_vec_arr = dom_vec.asarray()

        # operator
        for axis in range(space.ndim):
            partial = PartialDerivative(space, axis=axis, method=method,
                                        pad_mode=pad_mode,
                                        pad_const=pad_const)

            # Compare to helper function
            dx = space.cell_sides[axis]
            diff = finite_diff(dom_vec_arr, axis=axis, dx=dx, method=method,
                               pad_mode=pad_mode,
                               pad_const=pad_const)

            partial_vec = partial(dom_vec)
            assert all_almost_equal(partial_vec.asarray(), diff)

            # Test adjoint operator
            derivative = partial.derivative()
            ran_vec = noise_element(space)
            deriv_vec = derivative(dom_vec)
            adj_vec = derivative.adjoint(ran_vec)
            lhs = ran_vec.inner(deriv_vec)
            rhs = dom_vec.inner(adj_vec)

            # Check not to use trivial data
            assert lhs != 0
            assert rhs != 0
>           assert almost_equal(lhs, rhs)
E           assert almost_equal(-0.0015717172622680667, -0.0015716743469238283)

test\discr\diff_ops_test.py:281: AssertionError
```

clearly we are a bit overzealous on the accuracy for CUDA.

--- a/test/discr/diff_ops_test.py
+++ b/test/discr/diff_ops_test.py
@@ -278,7 +278,7 @@ def test_part_deriv(space, method, padding):
         # Check not to use trivial data
         assert lhs != 0
         assert rhs != 0
-        assert almost_equal(lhs, rhs)
+        assert almost_equal(lhs, rhs, places=4)
 
 
 # --- Gradient --- #
@@ -325,7 +325,7 @@ def test_gradient(space, method, padding):
     # Check not to use trivial data
     assert lhs != 0
     assert rhs != 0
-    assert almost_equal(lhs, rhs)
+    assert almost_equal(lhs, rhs, places=4)
 
     # higher dimensional arrays
     lin_size = 3
@@ -386,7 +386,7 @@ def test_divergence(space, method, padding):
     # Check not to use trivial data
     assert lhs != 0
     assert rhs != 0
-    assert almost_equal(lhs, rhs)
+    assert almost_equal(lhs, rhs, places=4)
 
     # Higher dimensional arrays
     for ndim in range(1, 6):
@@ -443,7 +443,7 @@ def test_laplacian(space, padding):
     # Check not to use trivial data
     assert lhs != 0
     assert rhs != 0
-    assert almost_equal(lhs, rhs)
+    assert almost_equal(lhs, rhs, places=4)
 
 
 if __name__ == '__main__':

